torch/distributions/poisson.py

```diff
@@ -51,7 +51,7 @@ def __init__(self, rate):
 
     def sample(self, sample_shape=torch.Size()):
         shape = self._extended_shape(sample_shape)
-        return _poisson(self.rate.expand(shape))
+        return _poisson(self.rate.expand(shape)).detach()
```

Comment:
I think all those `.detach()`s will raise error if parameters no distributions are not `Variable`

Reply:
@fritzo here

Reply:
Ok, I'll use the `hasattr(x, 'detach')` trick more consistently. If upstream reviewers say we can assume everything is a variable, I'll revert this and also `Distribution.sample()`.